content/client-lib-development-guide/features.textile

```diff
@@ -1590,27 +1603,22 @@ enum StatsIntervalGranularity:
 
 class DeviceDetails:
   id: String
-  platform: DevicePlatform
-  formFactor: DeviceFormFactor
   clientId: String?
+  formFactor: DeviceFormFactor
   metadata: JsonObject
-  updateToken: String
+  platform: DevicePlatform
   push: DevicePushDetails
+  updateToken: String
 
 class DevicePushDetails:
-  transportType: DevicePushTransportType
-  transportId: String
-  transportToken: String?
-  state: .Active | .Failing | .Failed
   errorReason: ErrorInfo?
-  metadata: JsonObject
+  recipient: JsonObject
+  state: .Active | .Failing | .Failed
 
 class LocalDevice extends DeviceDetails:
-  resetId()
-  resetUpdateToken() => io
+  resetUpdateToken(token: String?) => io
```

Comment:
Ok, but if we are to have separate methods, then I would argue `resetUpdateToken` is  misleading regardless of which request you meant to make.

Should it not perhaps be:

`setLocalUpdateToken(updateToken: String)`
and 
`reissueUpdateToken() => io`

The reason I went for `setLocal..` as opposed to just `setUpdateToken` is to attempt to make it clear that this update only affects the LocalDevice.  Saying that, perhaps that's a bit unnecessary given the object is a `LocalDevice` anyway?